plugin/trino-hive/src/main/java/io/trino/plugin/hive/SkipGlacierObjects.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive;
+
+import java.util.Set;
+
+public enum SkipGlacierObjects {
+    READ_ALL,
+    READ_NON_GLACIER,
+    READ_NON_GLACIER_AND_RESTORED;
+
+    /**
+     * Checks if the S3 object is not an object with a storage class of glacier/deep_archive
+     *
+     * @param tags set of tags stating glacier and/or restored
+     * @return If the S3 object is not an object with a storage class of glacier/deep_archive
+     */
+    private static boolean isNotGlacierObject(Set<String> tags)
+    {
+        return !tags.contains("s3:glacier");
+    }
+
+    /**
+     * Only restored objects will have the restoreExpiryDate set.
+     * Ignore not-restored objects and in-progress restores.
+     *
+     * @param tags set of tags stating glacier and/or restored
+     * @return If the S3 object is completely restored
+     */
+    private static boolean isCompletedRestoredObject(Set<String> tags)
+    {
+        return tags.contains("s3:glacier") || tags.contains("s3:glacierRestored");
+    }
+
+    public boolean shouldReadObject(Set<String> tags)
+    {
+        return switch (this) {
+            case READ_ALL -> true;
+            case READ_NON_GLACIER -> isNotGlacierObject(tags);
+            case READ_NON_GLACIER_AND_RESTORED -> isCompletedRestoredObject(tags);
```

Comment:
Shouldn't this be `isNotGlacierObject(tags) || isCompletedRestoredObject(tags)`?